### DISCUSSION TITLE
Why not merge the Object Node and Access modules into one module?

### DISCUSSION THREAD

**Discussion by FourSpaces** [cubefs/cubefs#3405] [Q&A] [answered]
Why not merge the Object Node and Access modules into one module?

The advantages of combining the Object Node and Access modules into one module:

> 1. Reduce the network connection overhead from Object Node to Access once, and further reduce read and write latency.
> 2. It is also possible to remove dependencies on consult components, reduce component size, and reduce maintenance costs.

**Comment by Cloudstriff** [comment, marked as answer]
We are on the way, we will first support the access SDK functionality, and then integrate the SDK functionality through the Object Node.
